### PR TITLE
pulumi: added support for aarch64-darwin and added script to help fetch latest plugins version

### PR DESCRIPTION
**Pull request by dbarrosop**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by austinbutler** [comment]
CC @t0yv0

**Comment by austinbutler** [comment]
Result of `nixpkgs-review pr 134624` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pulumi-bin</li>
  </ul>
</details>

File: pkgs/tools/admin/pulumi/get_latest_plugins.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+plugins=(
```
Comment (austinbutler) [inline pkgs/tools/admin/pulumi/get_latest_plugins.sh:1]:
```suggestion
#!/usr/bin/env bash

plugins=(
```

Comment (austinbutler) [reply pkgs/tools/admin/pulumi/get_latest_plugins.sh:1]: Can you make the script executable?

**Review by austinbutler — CHANGES_REQUESTED** [review]

File: pkgs/tools/admin/pulumi/get_latest_plugins.sh (hunk 1 above)
Comment (dbarrosop) [reply pkgs/tools/admin/pulumi/get_latest_plugins.sh:3]: Added shebang and made the script executable. Thanks!

**Comment by t0yv0** [comment]
Just installed pulumi-bin from these definitions on my Mac and smoke-tests seem fine.

**Comment by prusnak** [comment]
Superseded by https://github.com/NixOS/nixpkgs/pull/137223

**Pull request closed without merging**